File: tests/syntax/supported/clojure/incremental.clj
(ns app.handler)

(defn handle [request]
  (let [method (:method request)
        path (:uri request)]
    (case method
      :get {:status 200 :body path}
      :post {:status 201}
      {:status 405})))

(def routes
  {"/health" handle
   "/status" handle})

File: tests/syntax/supported/elixir/incremental.ex
defmodule Router do
  def handle(request) do
    method = request.method
    path = request.path

    case method do
      :get -> {200, path}
      :post -> {201, ""}
      _ -> {405, ""}
    end
  end
end

routes = [
  {"/health", &Router.handle/1},
  {"/status", &Router.handle/1}
]
